torch/_subclasses/meta_utils.py

```diff
@@ -239,10 +239,10 @@ def sym_sizes_strides_storage_offset(t):
         if self.get_tensor_memo(t) is None:
             with torch.inference_mode(t.is_inference()):
                 if t.is_sparse:
-                    # TODO: Delete this assert, and just attempt making the
-                    # sparse tensor anyway; even if there is a shape_env, this
-                    # tensor might be all static
-                    assert shape_env is None, "symbolic on sparse NYI"
+                    static = all(isinstance(s, int) for s in t.size()) and all(
+                        isinstance(s, int) for s in t.stride()
+                    )
```

Comment:
use `free_symbols` from torch.fx.experimental.symbolic_shapes. In particular, that will handle symbolic storage offset correctly too